lib/msf/core/handler/reverse_http.rb

```diff
@@ -388,14 +388,20 @@ def on_request(cli, req)
           begin
             blob = self.generate_stage(url: url, uuid: uuid, uri: conn_id)
             blob = encode_stage(blob) if self.respond_to?(:encode_stage)
+            # remove this when we make http payloads prepend stage sizes by default
+            if defined?(read_stage_size?) && read_stage_size?
+              print_status("Appending Stage Size For HTTP[S]...")
+              blob = [ blob.length ].pack('V') + blob
+            end
 
             print_status("Staging #{uuid.arch} payload (#{blob.length} bytes) ...")
 
             resp['Content-Type'] = 'application/octet-stream'
             resp.body = blob
 
-          rescue NoMethodError
-            print_error("Staging failed. This can occur when stageless listeners are used with staged payloads.")
+          rescue NoMethodError => e
+            print_error("Staging failed. This can occur when stageless listeners are used with staged payloads.\n#{e}")
+            elog(e)
```

Comment:
```suggestion
          rescue NoMethodError => e
            print_error("Staging failed. This can occur when stageless listeners are used with staged payloads.")
            elog("Staging failed. This can occur when stageless listeners are used with staged payloads.", error: e)
```